File: go/AddDigits/AddDigits.go
package main

import (
	"fmt"
	"strconv"
)

// Time -> 0(N^2) Space -> 0(N)
func addDigits(num int) int {
	for len(fmt.Sprintf("%v", num)) != 1 {
		runSum := 0
		val := fmt.Sprintf("%v", num)
		for idx := range val {
			add, _ := strconv.Atoi(string(val[idx]))
			runSum += add
		}
		num = runSum
	}
	return num
}

func main() {
	fmt.Println(addDigits(121))
}

File: python/addDigits.py
# Time -> 0(N^2) Space -> 0(N)
def addDigits(num: int) -> int:
    while len(str(num)) != 1:
        run_sum = 0
        val = str(num)
        for v in val:
            run_sum += int(v)
        num = run_sum

    return num


print(addDigits(121))
